src/henplus/HenPlus.java

```
@@ -339,7 +339,22 @@ public String readlineFromFile() throws IOException {
     private void storeLineInHistory() {
         final String line = _historyLine.toString();
         if (!"".equals(line) && !line.equals(_previousHistoryLine)) {
-            Readline.addToHistory(line);
+            try {
+                Readline.addToHistory(line);
+            } catch (Exception uee) {
+                // This error might happen if HenPlus is running
+                // on a system which doesn't have full encoding support --
+                // such as Alpine linux -- and the query (or query results?)
+                // include non-ASCII characters. This error will essetinally
+                // kill HenPlus, but it doesn't really affect the output,
+                // so we will ignore it.
+
+                // Notice:
+                // The exception thrown is really java.io.UnsupportedEncoding
+                // but due to it being thrown from the native code, it's not
+                // declared properly on the java interface so we can't
+                // specifically catch it here.
+            }
```

Could the caught exception be runtime-tested to be an UnsupportedEncoding exception in order to make sure not to swallow anything else?